lib/hathiRecord.py

```diff
@@ -0,0 +1,590 @@
+        if self.modified is None:
+            self.modified = datetime.now().strftime('%Y-%m-%d %H:%M:%S')
+            logger.debug('Assigning generated timestamp of {} to new record'.format(self.modified))
+        elif type(self.modified) is datetime:
+            self.modified = self.modified.strftime('%Y-%m-%d %H:%M:%S')
+
+    def __repr__(self):
+        return '<Hathi(title={})>'.format(self.work.title)
+
+    def buildWork(self):
+        """Construct the SFR Work object from the Hathi data"""
+        self.work.title = self.ingest['title']
+        self.work.series = self.ingest['description']
+        logger.info('Creating work record for {}'.format(self.work.title))
+        # The primary identifier for this work is a HathiTrust bib reference
+        self.work.primary_identifier = Identifier(
+            type='hathi',
+            identifier=self.ingest['bib_key'],
+            weight=1
+        )
+        logger.debug('Setting primary_identifier to {}'.format(
+            self.work.primary_identifier
+        ))
+
+        # Identifiers are included across multiple columns
+        workIdentifiers = [
+            ('hathi', 'bib_key'),
+            ('hathi', 'htid'),
+            (None, 'source_id'),
+            ('isbn', 'isbns'),
+            ('issn', 'issns'),
+            ('lccn', 'lccns'),
+            ('oclc', 'oclcs')
+        ]
+        for idType, key in workIdentifiers:
+            logger.debug('Setting identifiers {}'.format(idType))
+            self.parseIdentifiers(self.work, idType, key)
+
+        # All government documents should be in the public_domain.
+        self.parseGovDoc(self.ingest['gov_doc'])
+
+        # The copyright date assigned to the work by HathiTrust
+        self.work.addClassItem('dates', Date, **{
+            'display_date': self.ingest['copyright_date'],
+            'date_range': self.ingest['copyright_date'],
+            'date_type': 'copyright_date'
+        })
+        logger.debug('Setting copyright date to {}'.format(
+            self.ingest['copyright_date']
+        ))
+
+        self.parseAuthor(self.ingest['author'])
+
+    def buildInstance(self, countryCodes):
+        """Constrict an instance record from the Hathi data provided. As
+        structured Hathi trust data will always correspond to a single
+        instance. A wok in Hathi can have multiple items, and this relationship
+        is reflected in the data.
+
+        We do not attempt to merge records at this phase, but will associated
+        works and instances related by identifiers when stored in the database.
+        """
+        self.instance.title = self.ingest['title']
+        self.instance.language = self.ingest['language']
+
+        logger.info('Creating instance record for work {}'.format(self.work))
+
+        self.parsePubPlace(self.ingest['pub_place'], countryCodes)
+
+        # Instances and works will share many identifiers
+        instanceIdentifiers = [
```

Comment:
Should this be placed at the top of the file where some other lists are constructed? The identical list seems to be used in `buildWork()`

Reply:
Yes, I was anticipating these being different, but they're not.